### PR TITLE
[FLINK-19247][docs-zh] Update Chinese documentation after removal of Kafka 0.10 and 0.11

### PR DESCRIPTION
**Pull request by Shawn-Hx**
## What is the purpose of the change

Update Chinese documentation after removal of Kafka 0.10 and 0.11.

## Brief change log

- Update Chinese documentation according to [FLINK-19152](https://github.com/apache/flink/pull/13360).
- Make the Chinese documentation to keep sync with the newest [Apache Kafka Connector](https://ci.apache.org/projects/flink/flink-docs-master/dev/connectors/kafka.html) documentation.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by Shawn-Hx** [comment]
Hi, @klion26 
This PR is ready to be reviewed. Could you help to review it at your convenience?
Thanks~

File: docs/dev/connectors/kafka.zh.md [hunk 1]
```diff
@@ -302,8 +217,6 @@ Flink Kafka Consumer 支持发现动态创建的 Kafka 分区，并使用精准
 
 默认情况下，是禁用了分区发现的。若要启用它，请在提供的属性配置中为 `flink.partition-discovery.interval-millis` 设置大于 0 的值，表示发现分区的间隔是以毫秒为单位的。
 
-<span class="label label-danger">局限性</span> 当从 Flink 1.3.x 之前的 Flink 版本的 savepoint 恢复 consumer 时，分区发现无法在恢复运行时启用。如果启用了，那么还原将会失败并且出现异常。在这种情况下，为了使用分区发现，请首先在 Flink 1.3.x 中使用 savepoint，然后再从 savepoint 中恢复。
-
 #### Topic 发现
```
Comment (klion26) [inline docs/dev/connectors/kafka.zh.md:220]: 既然更像这个文档了，也按照 [wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) 的方式，给所有标题添加一下 <a> 标签吧，可以仿照英文版的 url 进行添加

File: docs/dev/connectors/kafka.zh.md [hunk 2]
```diff
@@ -431,39 +350,42 @@ stream.addSink(myProducer);
...
- * *提供自定义属性*：producer 允许为内部 `KafkaProducer` 提供自定义属性配置。有关如何配置 Kafka Producer 的详细信息，请参阅  [Apache Kafka 文档](https://kafka.apache.org/documentation.html)。
- * *自定义分区器*：要将消息分配给特定的分区，可以向构造函数提供一个 `FlinkKafkaPartitioner` 的实现。这个分区器将被流中的每条记录调用，以确定消息应该发送到目标 topic 的哪个具体分区里。有关详细信息，请参阅 [Kafka Producer 分区方案](#kafka-producer-分区方案)。
- * *高级的序列化 schema*：与 consumer 类似，producer 还允许使用名为 `KeyedSerializationSchema` 的高级序列化 schema，该 schema 允许单独序列化 key 和 value。它还允许覆盖目标 topic，以便 producer 实例可以将数据发送到多个 topic。
+Flink Kafka Producer 需要知道如何将 Java/Scala 对象转化为二进制数据。
 
-### Kafka Producer 分区方案
+`KafkaSerializationSchema` 允许用户指定这样的 schema。它会为每个记录调用 `ProducerRecord<byte[], byte[]> serialize(T element, @Nullable Long timestamp)` 方法，产生一个写入到 Kafka 的 `ProducerRecord`。
 
-默认情况下，如果没有为 Flink Kafka Producer 指定自定义分区程序，则 producer 将使用 `FlinkFixedPartitioner` 为每个 Flink Kafka Producer 并行子任务映射到单个 Kafka 分区（即，接收子任务接收到的所有消息都将位于同一个 Kafka 分区中）。
+用户可以对如何将数据写到Kafka进行细粒度的控制。你可以通过 producer record：
```
Comment (klion26) [inline docs/dev/connectors/kafka.zh.md:373]:
```suggestion
用户可以对如何将数据写到 Kafka 进行细粒度的控制。你可以通过 producer record：
```

File: docs/dev/connectors/kafka.zh.md [hunk 3]
```diff
@@ -431,39 +350,42 @@ stream.addSink(myProducer);
...
-
-也可以完全避免使用分区器，并简单地让 Kafka 通过其附加 key 写入的消息进行分区（使用提供的序列化 schema 为每条记录确定分区）。
-为此，在实例化 producer 时提供 `null` 自定义分区程序，提供 `null` 作为自定义分区器是很重要的; 如上所述，如果未指定自定义分区程序，则默认使用 `FlinkFixedPartitioner`。
+* 设置 header 值
+* 为每个 record 定义 key
+* 指定数据的自定义分区
 
 ### Kafka Producer 和容错
 
+启用 Flink 的checkpointing 后，`FlinkKafkaProducer` 可以提供精确一次的语义保证。
```
Comment (klion26) [inline docs/dev/connectors/kafka.zh.md:381]:
```suggestion
启用 Flink 的 checkpointing 后，`FlinkKafkaProducer` 可以提供精确一次的语义保证。
```

File: docs/dev/connectors/kafka.zh.md [hunk 4]
```diff
@@ -512,6 +434,17 @@ Flink 通过 Kafka 连接器提供了一流的支持，可以对 Kerberos 配置
 
 有关 Kerberos 安全性 Flink 配置的更多信息，请参见[这里]({{ site.baseurl }}/zh/ops/config.html)。你也可以在[这里]({{ site.baseurl }}/zh/ops/security-kerberos.html)进一步了解 Flink 如何在内部设置基于 kerberos 的安全性。
 
+## 升级到最近的连接器版本
+
+通用的升级步骤概述见 [升级 Jobs 和 Flink 版本指南]({{ site.baseurl }}/ops/upgrading.html)。对于 Kafka，你还需要遵循这些步骤：
```
Comment (klion26) [inline docs/dev/connectors/kafka.zh.md:439]: 链接替换成 `{% link %}`

**Review by klion26 — COMMENTED** [review]
@Shawn-Hx thanks for your contribution, left some comments, please take a look

File: docs/dev/connectors/kafka.zh.md [hunk 5]
```diff
@@ -23,90 +23,32 @@ specific language governing permissions and limitations
...
-* 不要同时升级 Flink 和 Operator。
-* 确保你的 Job 中所使用的 Kafka Consumer 和 Kafka Producer 分配了唯一的标识符（uid）。
-* 使用 stop with savepoint 的特性来执行 savepoint（例如，使用 `stop --withSavepoint`）[CLI 命令]({{ site.baseurl }}/zh/ops/cli.html)。
-
-### 用法
-
-要使用通用的 Kafka 连接器，请为它添加依赖关系：
+Apache Flink 集成了通用的 Kafka 连接器，它会尽力与 Kafka client 的最新版本保持同步。
+该连接器使用的 Kafka client 版本可能会在 Flink 版本之间发生变化。
+当前 Kafka client 向后兼容 0.10.0 或更高版本的 Kafka broker。
```
[the remark this answers is not in the document]
Comment (Shawn-Hx) [reply docs/dev/connectors/kafka.zh.md:36]: 这里我不是很确定。
英文文档中的原文是："Modern Kafka clients are backwards compatible with broker versions 0.10.0 or later"。"backwards compatible" 直译的话应该是 “向后兼容” ？

**Comment by Shawn-Hx** [comment]
Hi, @klion26 
Thanks for your good suggestions !
I have make some changes according to your advice. Please take a look.

File: docs/dev/connectors/kafka.zh.md [hunk 6]
```diff
@@ -431,39 +371,46 @@ stream.addSink(myProducer);
...
 
+启用 Flink 的 checkpointing 后，`FlinkKafkaProducer` 可以提供精确一次的语义保证。
+
+除了启用 Flink 的 checkpointing，你也可以通过将适当的 `semantic` 参数传递给 `FlinkKafkaProducer` 来选择三种不同的操作模式：
+
+* `Semantic.NONE`：Flink 不会有任何语义的保证，产生的记录可能会丢失或重复。
+* `Semantic.AT_LEAST_ONCE`（默认设置）：可以保证不会丢失任何记录（但是记录可能会重复）
+* `Semantic.EXACTLY_ONCE`：使用 Kafka 事务提供精确一次语义。无论何时，在使用事务写入 Kafka 时，都要记得为所有消费 Kafka 消息的应用程序设置所需的 `isolation.level`（`read_committed` 或 `read_uncommitted` - 后者是默认值）。
+
 ##### 注意事项
```
Comment (klion26) [inline docs/dev/connectors/kafka.zh.md:414]: 这里是不是漏了 <a> 标签

Comment (Shawn-Hx) [reply docs/dev/connectors/kafka.zh.md:416]: Thanks for your reminding. Have added the tag.

**Comment by Shawn-Hx** [comment]
Hi, @klion26 .
I have made changes according to your advice. Please take a look.
Thanks~

**Review by klion26 — APPROVED** [review]
@Shawn-Hx thanks for the update. LGTM now
cc @wuchong @rmetzger

**Pull request merged**